File: Day244/rotateBits.go
package main

import (
	"fmt"
	"math"
)

func rotate(n int, d int ) []int {
	answer := make([]int, 0)
	d %= 16
	answer = append(answer, ((n<<d) | (n<<(16-d))) & (int(math.Pow(2, 16))-1))
	answer = append(answer, ((n>>d) | (n>>(16-d))) & (int(math.Pow(2, 16))-1))
	return answer
}

func main() {
	fmt.Println(rotate(28, 2))
	fmt.Println(rotate(18, 2))
	fmt.Println(rotate(234, 34))
}

File: Day244/rotateBits.java
package Day244;

import java.util.List;
import java.util.ArrayList;

public class rotateBits {

    public static List<Integer> rotate(int n, int d) {
        List<Integer> list = new ArrayList<>();
        d = d % 16;
        System.out.println((n<<d) + " "+(n>>(16-d))+" "+ (Math.pow(2,16) -1) );
        System.out.println((n<<d) | (n>>(16-d)));
        System.out.println((n>>d) + " "+(n<<(16-d))+" "+ (Math.pow(2,16) -1) );
        System.out.println((n>>d) | (n<<(16-d)));
        int a = ((n<<d) | (n>>(16-d)) & (int)Math.pow(2, 16)-1);
        int b = ((n>>d) | (n<<(16-d)) & (int)Math.pow(2, 16)-1);
        list.add(a);
        list.add(b);
        return list;
    }

    public static void main(String[] args) {
        System.out.println(rotate(18, 2));
        System.out.println(rotate(8, 2));
        System.out.println(rotate(2, 3));
        System.out.println(rotate(28, 2));
    }
}